src/main/java/org/opentripplanner/updater/vehicle_rental/datasources/GbfsFeedLoader.java

```diff
@@ -0,0 +1,159 @@
+package org.opentripplanner.updater.vehicle_rental.datasources;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.entur.gbfs.v2_2.gbfs.GBFS;
+import org.entur.gbfs.v2_2.gbfs.GBFSFeed;
+import org.entur.gbfs.v2_2.gbfs.GBFSFeedName;
+import org.entur.gbfs.v2_2.gbfs.GBFSFeeds;
+import org.opentripplanner.util.HttpUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.InvocationTargetException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Class for managing the state and loading of complete GBFS datasets, and updating them according to individual feed's
+ * TTL rules.
+ */
+public class GbfsFeedLoader {
+    private static final Logger LOG = LoggerFactory.getLogger(GbfsFeedLoader.class);
+
+    private static final ObjectMapper objectMapper = new ObjectMapper();
+
+    /** One updater per feed type(?)*/
+    private final Map<GBFSFeedName, GBFSFeedUpdater<?>> feedUpdaters = new HashMap<>();
+
+    private final Map<String, String> httpHeaders;
+
+    public GbfsFeedLoader(String url, Map<String, String> httpHeaders, String languageCode) {
+        this.httpHeaders = httpHeaders;
+        URI uri;
+        try {
+            uri = new URI(url);
+        } catch (URISyntaxException e) {
+            throw new RuntimeException("Invalid url " + url);
+        }
+
+        // Fetch autoconfiguration file
+        GBFS data = fetchFeed(uri, httpHeaders, GBFS.class);
+        if (data == null) {
+            throw new RuntimeException("Could not fetch feed " + uri + " autoconfiguration file");
+        }
+
+        // Pick first language if none defined
+        GBFSFeeds feeds = languageCode == null
+                ? data.getFeedsData().values().iterator().next()
+                : data.getFeedsData().get(languageCode);
+        if (feeds == null) {
+            throw new RuntimeException("Language " + languageCode + " does not exist in feed " + uri);
+        }
+
+        // Create updater for each file
+        for (GBFSFeed feed : feeds.getFeeds()) {
+            GBFSFeedName feedName = feed.getName();
+            if (feedUpdaters.containsKey(feedName)) {
+                throw new RuntimeException("Feed " + url + " contains duplicate url for feed " + feedName);
```

Comment:
`"Feed contains duplicate url for feed " + feedName + ". Urls: " + url + ", " + feedUpdaters.get(feedName).url`

(Unsure if the `", " + feedUpdaters.get(feedName).url` is relevant or even compile )